# Astma kasalligi bilan kasallangan bemorlar tahlili

In [1]:
import pandas as pd

In [4]:
df = pd.read_csv("asthma_disease_data.csv")
df

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0             0.894448     5.488696      8.701003  ...          1.369051   
1             5.897329     6.341014      5.153966  ...          2.197767   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
4             4.604493     3.127048      9.625799  ...          3.470589   
...                ...          ...           ...  ...               ...   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2388          5.805180     4.386992      7.731192  ...          1.132977   
2389          4.735169     8.214064      7.483521  ...          1.685962   
2390          9.672637     7.362861      6.717272  ...          3.481549   
2391          1.613138     7.412878      8.512253  ...          2.280613   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            4.941206         0                  0               1         0   
1            1.702393         1                  0               0         1   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
4            3.067944         1                  1               1         0   
...               ...       ...                ...             ...       ...   
2387         5.166032         0                  1               0         0   
2388         5.509502         0                  0               0         1   
2389         3.346877         1                  0               1         1   
2390         1.713274         0                  1               1         0   
2391         2.453284         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                     0                1          0       Dr_Confid  
1                     1                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
4                     0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2387                  0                1          1       Dr_Confid  
2388                  1                0          1       Dr_Confid  
2389                  0                1          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  
2391                  1                0          0       Dr_Confid  

[2392 rows x 29 columns]

Patient ID: from 5034 to 7425, Bemorlarni ID si
Age: from 5 to 80 years
Gender: 0 - Male, 1 - Female
EthnicityL: 0: Caucasian
            1: African American
            2: Asian
            3: Other
Education level: 0: None
                 1: High School
                 2: Bachelor's
                 3: Higher
BMI: Body mass index. from 15 to 40 (18.5 to 24.9 - healthy, 25.0 to 29.9 - overweight, 30.0 or higher - obese)
Smoking: 1 - Yes, 0 - No
PhysicalActivity: Weekly physical activity in hours, ranging from 0 to 10.
DietQuality: Diet quality score, ranging from 0 to 10.
SleepQuality: Sleep quality score, ranging from 4 to 10.

I worked with this indicators!

### Astma kasalligi bilan kasallanganlarni o'rtacha yoshi va Body mass indexi

In [7]:
df[["Age", "BMI"]].mean()

Age    42.137960
BMI    27.244877
dtype: float64

### Astma bilan kasallangan odamlarning minimum va maksimum yoshlari va body mass indexlari

In [6]:
df[["Age", "BMI"]].min()

Age     5.000000
BMI    15.031803
dtype: float64

### Astma bilan kasallanganlarning modasi(Eng kop takrorlanuvchi yoshi)

In [12]:
df["Age"].mode()

0    51
Name: Age, dtype: int64

In [14]:
# Standard og'ish(yosh va BMI bo'yicha bo'yicha)

df[["Age", "BMI"]].std()

Age    21.606655
BMI     7.201628
dtype: float64

In [16]:
# yuqorida berilgan barcchasini describe methodi bilan korib olish
df[["Age", "BMI"]].describe()

Age          BMI
count  2392.000000  2392.000000
mean     42.137960    27.244877
std      21.606655     7.201628
min       5.000000    15.031803
25%      23.000000    20.968313
50%      42.000000    27.052202
75%      61.000000    33.555903
max      79.000000    39.985611

### Nechi yoshli bemorlar nechtadan ekani haqida summary

In [17]:
df["Age"].value_counts()

Age
51    47
12    45
27    43
57    42
22    40
      ..
52    24
68    23
30    23
55    22
9     21
Name: count, Length: 75, dtype: int64

### Ayollar va erkaklarni soni 0 - male, 1 - female

In [19]:
df["Gender"].value_counts()

Gender
0    1212
1    1180
Name: count, dtype: int64

In [20]:
female_mask = df["Gender"] == 1
male_mask = df["Gender"] == 0

female_df = df[female_mask]
male_df = df[male_mask]

female_df

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
1          5035   26       1          2               2  22.757042        0   
3          5037   40       1          2               1  38.515278        0   
6          5040   45       1          1               1  30.245954        1   
8          5042   49       1          1               2  32.676204        0   
9          5043   45       1          1               1  29.910298        0   
...         ...  ...     ...        ...             ...        ...      ...   
2385       7419   19       1          1               2  37.913891        0   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
1             5.897329     6.341014      5.153966  ...          2.197767   
3             1.404503     5.826532      4.253036  ...          3.032037   
6             9.371784     7.030507      5.746128  ...          2.995100   
8             2.690256     3.920034      5.843645  ...          1.761242   
9             2.895720     2.607700      7.234908  ...          2.848420   
...                ...          ...           ...  ...               ...   
2385          5.595540     3.120986      4.122047  ...          3.849256   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2388          5.805180     4.386992      7.731192  ...          1.132977   
2390          9.672637     7.362861      6.717272  ...          3.481549   
2391          1.613138     7.412878      8.512253  ...          2.280613   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
1            1.702393         1                  0               0         1   
3            2.300159         1                  0               1         1   
6            1.701512         1                  1               1         0   
8            5.190931         1                  1               1         1   
9            5.771022         1                  0               0         0   
...               ...       ...                ...             ...       ...   
2385         4.910336         0                  1               1         1   
2387         5.166032         0                  1               0         0   
2388         5.509502         0                  0               0         1   
2390         1.713274         0                  1               1         0   
2391         2.453284         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
1                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
6                     0                0          0       Dr_Confid  
8                     0                0          0       Dr_Confid  
9                     1                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2385                  0                1          0       Dr_Confid  
2387                  0                1          1       Dr_Confid  
2388                  1                0          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  
2391                  1                0          0       Dr_Confid  

[1180 rows x 29 columns]

In [25]:
df[female_mask]["Smoking"].value_counts(normalize=True)

Smoking
0    0.862712
1    0.137288
Name: proportion, dtype: float64

In [26]:
df[male_mask]["Smoking"].value_counts(normalize=True)


Smoking
0    0.85396
1    0.14604
Name: proportion, dtype: float64

In [27]:
df["Ethnicity"].value_counts()

Ethnicity
0    1465
1     475
2     229
3     223
Name: count, dtype: int64

## Guruhlashtirib tahlil qilish

In [35]:
df.groupby(by = ["Gender"])["Ethnicity"]

In [43]:
symptoms_mask = ["Wheezing"] == 1
df.groupby(by = ["Gender"])[[symptoms_mask]].value_counts()


KeyError: 'Columns not found: False'